src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,429 @@
+ * sent downstream (default behavior of KStreams) but also the table side (not supported yet in Kafka Streams out of the
+ * box).  The example will delay join output for a configurable amount of time when a record arrives in the stream but
+ * doesn't yet have a matching record in the table.  If data happens to arrive in time on the table side, a "full" join
+ * output will be produced.  If table data does not arrive in time, then (like the default behavior) a join output will
+ * be sent where the table-side data is `null`.
+ *
+ * The approach in this example shares state stores between a stream-side and a table-side transformer.  This is safe
+ * because, if shared, Kafka Streams will place the transformers as well as the state stores into the same stream task,
+ * in which all access is exclusive and single-threaded.  The state store on the table side is the normal store of a
+ * KTable (thus avoiding data duplication due to store usage), whereas the state store on the stream side is manually
+ * added and attached to the processing topology.
+ *
+ * An alternative, more flexible approach is outlined in the code comments below, in case you need additional control
+ * over the join behavior, e.g. by including stream-side vs. table-side timestamps in the decision-making logic.
+ *
+ * The default stream-table join behavior of Kafka Streams (below: left join; inner join is similar) only triggers
+ * join output when data arrives at the stream side.
+ * See https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Streams+Join+Semantics.
+ *
+ * Kafka Streams INNER stream-table join:
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | -
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ * Kafka Streams LEFT stream-table join:
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | ("alice", (999.99, null)
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ *
+ * The code in this example changes the above behavior so that an application will wait a configurable amount of time
+ * for data to arrive also on the table before it produces the join output for a given key (here: "alice").  Depending
+ * on your use case, you might prefer this changed behavior over the default join semantics of Kafka Streams.
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | -
+ * 20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
```

Comment:
I changed the example to event-time.

The core behavior, however, is working as intended.  (I agree with your comment, but the example's point is to deviate from the DSL semantics.)  See my response at https://github.com/confluentinc/kafka-streams-examples/pull/197#discussion_r255471467.

Reply:
If the `Time` column is even time, the example is still incorrect because a stream record with **event-time** 10 should not join with **future** table record of event time 20. Thus, I don't understand why the example is not in event-time?

If `Time` column is processing time, the records should indicate their event-timestamps to reason about the join.